classes/activate.php

```diff
@@ -16,24 +16,51 @@ class Object_Sync_Sf_Activate {
 
 	protected $wpdb;
 	protected $version;
-	protected $installed_version;
+	protected $slug;
+	protected $option_prefix;
+	protected $schedulable_classes;
+
+	private $installed_version;
 
 	/**
 	* Constructor which sets up activate hooks
 	*
 	* @param object $wpdb
 	* @param string $version
 	* @param string $slug
+	* @param string $option_prefix
+	* @param array $schedulable_classes
 	*
 	*/
-	public function __construct( $wpdb, $version, $slug ) {
-		$this->wpdb              = $wpdb;
-		$this->version           = $version;
-		$this->installed_version = get_option( 'object_sync_for_salesforce_db_version', '' );
-		register_activation_hook( dirname( __DIR__ ) . '/' . $slug . '.php', array( $this, 'php_requirements' ) );
-		register_activation_hook( dirname( __DIR__ ) . '/' . $slug . '.php', array( $this, 'wordpress_salesforce_tables' ) );
-		register_activation_hook( dirname( __DIR__ ) . '/' . $slug . '.php', array( $this, 'add_roles_capabilities' ) );
-		add_action( 'plugins_loaded', array( $this, 'wordpress_salesforce_update_db_check' ) );
+	public function __construct( $wpdb, $version, $slug, $option_prefix, $schedulable_classes ) {
```

Comment:
Since `__construct` is a public method and this is the 1.4 release (not a 2.0 backwards compatibility breaking release), the new parameters should be optional. We can't guarantee developers haven't invoked this method.